### PR TITLE
ci(circleci): added sonar and git hooks (CORE-4768)

### PR DESCRIPTION
**Pull request by xavidop**
**Fixes or implements CORE-4768**

### Brief description. What is this change?

Added sonarcloud and git branch hooks

### Checklist

- [x] title of PR reflects the branch name
- [x] all commits adhere to conventional commits
- [ ] appropriate tests have been written
- [ ] all the dependendencies are upgraded

### CODE REVIEW

**Comment by frankgu968** [comment]
Can we disable the PR-label -> preview dev env as a check? We should still let it run, but can we not show it as a check? This wasn't possible about 6 months ago when I was working on this, but Github may have enabled that feature now...

File: tests/smoke/smokeTestRunner.ts [hunk 1]
```diff
@@ -13,6 +11,7 @@ import path from 'path';
 
 import { ServiceManager } from '../../backend';
 import config from '../../config';
+import '../../envSetup';
```
Comment (frankgu968) [inline tests/smoke/smokeTestRunner.ts:14]: Can we standardize this to `@/envSetup`? same thing for the others above

File: config.ts [hunk 2]
```diff
@@ -1,7 +1,6 @@
-import './envSetup';
-
 import * as Common from '@voiceflow/common';
 
+import './envSetup';
```
Comment (frankgu968) [inline config.ts:3]: Can we use `@/envSetup`?

**Review by frankgu968 — CHANGES_REQUESTED** [review]

File: tests/smoke/smokeTestRunner.ts (hunk 1 above)
Comment (xavidop) [reply tests/smoke/smokeTestRunner.ts]: done!

File: config.ts (hunk 2 above)
Comment (xavidop) [reply config.ts]: done!

**Review by frankgu968 — APPROVED** [review]

**Pull request merged**